### PR TITLE
CBL-4023: Make SyncServerCollection tests compatible with SGW 3.1.0

### PR DESCRIPTION
**Pull request by callumbirks**
Tested with SGW 3.1.0-438, all tests are passing.

### CODE REVIEW

**Comment by callumbirks** [comment]
Realized the channel_access fixes are hardcoded for the "Roses" collection, working on fixing this now

File: Replicator/tests/SG.cc [hunk 1]
```diff
@@ -138,9 +138,37 @@ bool SG::deleteUser(const string &username) const {
 }
 
 bool SG::assignUserChannel(const std::string& username, const std::vector<std::string> &channelIDs) const {
```
Comment (jianminzhao) [inline Replicator/tests/SG.cc:140]: We probably need an argument telling what collections will be involved. Or we may want to test different channelIDs for different collections,

Comment (callumbirks) [reply Replicator/tests/SG.cc]: I believe the commit I just made should address those two concerns

**Comment by callumbirks** [comment]
The reason for using a multimap is to make parsing the multiple collections easier. With this method, the user should be able to provide collectionSpecs with different scopes and collections, and it will all work as expected.

**Review by pasin — APPROVED** [review]

File: Replicator/tests/SG.cc [hunk 2]
```diff
@@ -123,11 +124,10 @@ alloc_slice SG::addChannelToJSON(slice json, slice ckey, const std::vector<std::
 bool SG::createUser(const std::string& username, const std::string& password,
                     const std::vector<std::string> &channelIDs) const {
```
Comment (jianminzhao) [inline Replicator/tests/SG.cc:125]: Where is "channelIds" used in this function?

File: Replicator/tests/SG.hh [hunk 3]
```diff
@@ -50,7 +50,13 @@ public:
...
                     const std::vector<std::string> &channelIDs) const;
     bool deleteUser(const std::string& username) const;
-    bool assignUserChannel(const std::string& username, const std::vector<std::string>& channelIDs) const;
+    // Assign given channels to the user with given username, in the given collections
+    bool assignUserChannel(const std::string& username, const std::vector<C4CollectionSpec>& collectionSpecs, const std::vector<std::string>& channelIDs) const;
+    // Same as above, but accepts an array parameter
+    template <size_t N>
+    bool assignUserChannel(const std::string& username, const std::array<C4CollectionSpec, N>& collectionSpecs, const std::vector<std::string>& channelIDs) const {
+        return assignUserChannel(username, std::vector(collectionSpecs.begin(), collectionSpecs.end()), channelIDs);
+    }
```
Comment (jianminzhao) [inline Replicator/tests/SG.hh:59]: Why not the other way around? From vector to array, I guess, is simple.

File: Replicator/tests/SG.cc (hunk 2 above)
Comment (callumbirks) [reply Replicator/tests/SG.cc:125]: Ah you’re right, I’ve removed the use from this function now

**Review by velicuvlad — APPROVED** [review]

**Review by jianminzhao — APPROVED** [review]

**Pull request merged**